torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -273,46 +278,85 @@ def gather(
                 Must to be provided ONLY on ``dst`` rank.
                 Default: ``None``
         """
+        def shard_size(shard_md):
+            res = 1
+            for s in shard_md.shard_sizes:
+                res *= s
+            return res
```

Comment:
Can't we just do `math.prod(shard_md.shard_sizes)`?

Reply:
TIL